### ISSUE TITLE
xilem_web: iframe `src` attribute not set after update

### ISSUE THREAD

**Issue by casey** [linebender/xilem#518]
In this example, the `iframe` src attribute isn't set after clicking the button:

```rust
use xilem_web::{document_body, elements::html, interfaces::Element, App, DomView};

fn update(state: &mut bool) -> impl DomView<bool> {
    html::div((
        html::button("click me!").on_click(|state: &mut bool, _| *state = !*state),
        html::br(()),
        if *state {
            html::iframe(())
                .attr("src", "https://www.google.com")
                .boxed()
        } else {
            html::div("div!").boxed()
        },
    ))
}

fn main() {
    App::new(document_body(), false, update).run();
}
```

If the branches are swapped, the iframe `src` attribute is initially set, but unset after clicking the button twice.

**Comment by Philipp-M** [comment]
Thanks for reporting that bug, yeah indeed, I forgot to apply the attributes when the inner type changes... Should be fixed by #519

**Comment by casey** [comment]
Nice, that fixed it! Thank you!
